tests/test_routing_observation_store.py

```
@@ -0,0 +1,625 @@
+
+
+def test_concurrent_window_registration_keeps_longest_retention(tmp_path) -> None:
+    path = tmp_path / "routing.sqlite"
+    ready = threading.Barrier(2)
+    errors: list[BaseException] = []
+
+    def build(window_seconds: int) -> None:
+        try:
+            ready.wait(timeout=1.0)
+            SqliteRoutingObservationStore(path, window_seconds, clock=_Clock(100.0)).close()
+        except BaseException as exc:  # pragma: no cover - test assertion path
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `BaseException`

(BLE001)